src/sage/arith/misc.py

```diff
@@ -2045,18 +2065,34 @@ def xgcd(a, b):
         sage: xgcd(y^2, a*h*y + b)
         (1, 7*a^2/b^2, (((-h)*a)/b^2)*y + 1/b)
     """
-    try:
+    if b is not None:
+        # xgcd of two elements
+        try:
+            return a.xgcd(b)
+        except AttributeError:
+            a = py_scalar_to_element(a)
+            b = py_scalar_to_element(b)
+        except TypeError:
+            b = py_scalar_to_element(b)
         return a.xgcd(b)
-    except AttributeError:
-        a = py_scalar_to_element(a)
-        b = py_scalar_to_element(b)
-    except TypeError:
-        b = py_scalar_to_element(b)
-    return a.xgcd(b)
+    else:
+        # xgcd of >=2 elements
+        if not isinstance(a, (tuple, list)):
+            raise TypeError("Input `a` has to be a tuple or a list.")
+        if len(a) < 2:
+            raise ValueError("At least two integers or elements should be given.")
+        if len(a) == 2:
+            return xgcd(a[0], a[1])
+        else:  # Compute xgcd recursively
+            xgcd_t = xgcd(a[1:])  # xgcd_t[0] = sum(xgcd_t[i] * a[i], 1 \le i \le n-1)
```

Comment:
```suggestion
            # xgcd_t[0] = sum(xgcd_t[i] * a[i], 1 \le i \le n-1)
            xgcd_t = xgcd(a[1:])
```
I don't think it will be too difficult to implement this iteratively instead to avoid stack overflow. Do you mind doing that? If you follow the call trace then `xgcd([1,2,3,4,5])` calls `xgcd([2,3,4,5])` then ... then `xgcd([4,5])`, before any computation is done. So I think you can instead process `a` in reverse order in pairs and keep the same computation

Reply:
Updated!